reference/tools/intel.rst

```diff
@@ -1,6 +1,148 @@
 .. _reference_tools_intel:
 
+conan.tools.intel
+=================
+
 IntelCC
-=======
+-------
+
+This tool helps you to manage the new Intel oneAPI `DPC++/C++ <https://software.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top.html>`_ and
+`Classic <https://software.intel.com/content/www/us/en/develop/documentation/cpp-compiler-developer-guide-and-reference/top.html>`_ ecosystem in Conan.
+
+.. warning::
+
+    This generator is **experimental** and subject to breaking changes.
+
+.. warning::
+
+    macOS is not supported for the Intel oneAPI DPC++/C++ (icx/icpx or dpcpp) compilers. For macOS or Xcode support, you'll have to use the Intel C++ Classic Compiler.
+
+.. note::
+
+    Remember, you need to have installed previously the `Intel oneAPI software <https://www.intel.com/content/www/us/en/developer/tools/oneapi/toolkits.html>`_.
+
+
+
+This generator creates a ``conanintelsetvars.sh|bat`` wrapping the Intel script ``setvars.sh|bat`` that sets the Intel oneAPI
+environment variables needed. That script is the first step to start using the Intel compilers because it's setting some
+important variables in your local environment.
+
+In summary, the ``IntelCC`` generator:
+
+#. Reads your profile ``[settings]`` and ``[conf]``.
+#. Uses that information to generate a ``conanintelsetvars.sh|bat`` script with the command to load the Intel ``setvars.sh|bat`` script.
+#. Then, you or the chosen generator will be able to run that script and use any Intel compiler to compile the project.
+
+.. note::
+
+    You can launch the ``conanintelsetvars.sh|bat`` before calling your intel compiler to build a project.
+    Conan will also call it in the conanfile ``build(self)`` method when running any command with ``self.run``.
+
+
+At first, ensure you are using a *profile* like this one:
+
+.. code-block:: text
+    :caption: *intelprofile*
+
+    [settings]
+    ...
+    compiler=intel-cc
+    compiler.mode=dpcpp
+    compiler.version=2021.3
+    compiler.libcxx=libstdc++
+    build_type=Release
+
+    [buildenv]
+    CC=dpcpp
+    CXX=dpcpp
```

Comment:
Does this work on Windows too? Maybe something to discuss in the future if this default could/should be automated.

Reply:
Yes, it works. For sure!